src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -17903,5 +17925,890 @@ public class XAttribute : System.Attribute
+        public void MissingCtor_TypeParameter_IEnumerable_WithWrongGenericIEnumerable()
+        {
+            string source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                class C
+                {
+                    static T1 Create1<T1>() where T1 : IEnumerable, IEnumerable<string> => []; // 1
+                    static T2 Create2<T2>() where T2 : IEnumerable, IEnumerable<string>, new() => [];
+                    static T3 Create3<T3>() where T3 : struct, IEnumerable, IEnumerable<string> => [];
+                    static T4 Create4<T4>() where T4 : class, IEnumerable, IEnumerable<string> => []; // 2
+                }
+                """;
+
+            var comp = CreateCompilation(source).VerifyEmitDiagnostics(
+                // (6,76): error CS0304: Cannot create an instance of the variable type 'T1' because it does not have the new() constraint
+                //     static T1 Create1<T1>() where T1 : IEnumerable, IEnumerable<string> => []; // 1
+                Diagnostic(ErrorCode.ERR_NoNewTyvar, "[]").WithArguments("T1").WithLocation(6, 76),
+                // (9,83): error CS0304: Cannot create an instance of the variable type 'T4' because it does not have the new() constraint
+                //     static T4 Create4<T4>() where T4 : class, IEnumerable, IEnumerable<string> => []; // 2
+                Diagnostic(ErrorCode.ERR_NoNewTyvar, "[]").WithArguments("T4").WithLocation(9, 83)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var collections = tree.GetRoot().DescendantNodes().OfType<CollectionExpressionSyntax>().ToArray();
+
+            var conversion1 = model.GetConversion(collections[0]);
+            Assert.True(conversion1.IsValid);
+            Assert.True(conversion1.IsCollectionExpression);
+
+            var conversion2 = model.GetConversion(collections[1]);
+            Assert.True(conversion2.IsValid);
+            Assert.True(conversion2.IsCollectionExpression);
+
+            var typeInfo2 = model.GetTypeInfo(collections[1]);
+            Assert.Null(typeInfo2.Type);
+            Assert.Equal("T2", typeInfo2.ConvertedType.ToTestDisplayString());
+
+            var conversion3 = model.GetConversion(collections[2]);
+            Assert.True(conversion3.IsValid);
+            Assert.True(conversion3.IsCollectionExpression);
+
+            var typeInfo3 = model.GetTypeInfo(collections[2]);
+            Assert.Null(typeInfo3.Type);
+            Assert.Equal("T3", typeInfo3.ConvertedType.ToTestDisplayString());
+
+            var conversion4 = model.GetConversion(collections[3]);
+            Assert.True(conversion4.IsValid);
+            Assert.True(conversion4.IsCollectionExpression);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69521")]
+        public void MissingCtor_TypeParameter_WrongGenericIEnumerable()
```

Comment:
>MissingCtor_TypeParameter_WrongGenericIEnumerable [](http://example.com/codeflow?start=20&length=49)

Why is the generic `IEnumerable<T>` wrong? And is this test different from `MissingCtor_TypeParameter_GenericIEnumerable`?